src/App.js

```diff
@@ -1,11 +1,127 @@
-import React, { useState } from "react";
+import React, { useState, useRef, useEffect, useCallback } from "react";
 import styles from "./App.module.css";
+import "./index.css";
 
-function App() {
+const Tab = ({ children, selected = false, id, index, onClick }) => {
+  const ref = useRef(null);
+  useEffect(() => {
+    if (selected && ref.current) {
+      ref.current.focus();
+    }
+  }, [selected]);
 
   return (
-    <div className={styles.Wrapper}>
+    <button
+      role="tab"
+      type="button"
+      aria-selected={selected}
+      aria-controls={`${id}-tab`.split(" ").join("-")}
+      id={id}
+      tabIndex={selected ? undefined : -1}
+      onClick={() => onClick(index)}
+      ref={ref}
+    >
+      {children}
+    </button>
+  );
+};
+
+const TabList = ({ children, label, onKeyDown }) => (
+  <div role="tablist" aria-label={label} onKeyDown={onKeyDown}>
+    {children}
+  </div>
+);
+
+const TabPanel = ({ children, id, selected = false }) => (
+  <div
+    tabIndex={0}
+    role="tabpanel"
+    id={`${id}-tab-panel`.split(" ").join("-")}
```

Comment:
I don't think this logic `.split(" ").join("-")` belongs here. Move it higher in call tree